test/full_url_for.spec.js

```diff
@@ -12,13 +12,19 @@ describe('full_url_for', () => {
   it('internal url - root directory', () => {
     ctx.config.url = 'https://example.com';
     fullUrlFor('index.html').should.eql(ctx.config.url + '/index.html');
-    fullUrlFor('/').should.eql(ctx.config.url + '/');
+    fullUrlFor('/').should.eql(ctx.config.url);
   });
 
   it('internal url - subdirectory', () => {
     ctx.config.url = 'https://example.com/blog';
     fullUrlFor('index.html').should.eql(ctx.config.url + '/index.html');
-    fullUrlFor('/').should.eql(ctx.config.url + '/');
+    fullUrlFor('/').should.eql(ctx.config.url);
+  });
+
+  it('internal url - subdirectory with trailing slash', () => {
+    ctx.config.url = 'https://example.com/blog/';
```

Comment:
No. The case here just won't happen.

https://github.com/hexojs/hexo/blob/17ee23fa181bcc2c3b47eba5b345762aef1eac36/test/scripts/hexo/load_config.js#L93-L106

Just as the test case of `load_config` shows, the `url: https://hexo.io/` will result in `hexo.config.url.should.eql('https://hexo.io');`, while `root: foo` will result in `hexo.config.root.should.eql('foo/');`. Thus, the test case just won't exists.

Reply:
https://github.com/hexojs/hexo/blob/17ee23fa181bcc2c3b47eba5b345762aef1eac36/lib/hexo/load_config.js#L34-L35 should be removed if this PR is accepted. Those two lines are considered a workaround and this PR handles it.